src/east/system_commands/sys_setup.py

```diff
@@ -120,6 +122,54 @@ def _install_cppcheck(east: EastContext, exe_path: str):
     )
 
 
+def _get_clang_download_link():
+    """Just a convenience function for getting the link for the clang+llvm binaries.
+
+    This link will be updated regularly.
+    """
+
+    version = "16.0.0"
+
+    arch = platform.processor().lower()
+
+    if arch == "x86_64":
+        arch = "x86_64-linux-gnu-ubuntu-18.04"
+    elif arch == "aarch64":
+        arch = "aarch64-linux-gnu"
+    elif arch == "arm64":
+        arch = "arm64-apple-darwin"
+    else:
+        print(
+            f"Unsupported architecture ({arch}), East will download x86_64 version "
+            "of clang+llvm binaries."
+        )
+        arch = "x86_64-linux-gnu-ubuntu-18.04"
+
+    link = (
+        "https://github.com/llvm/llvm-project/releases/download/"
+        f"llvmorg-{version}/clang+llvm-{version}-{arch}.tar.xz"
+    )
+
+    return link
+
+
+def _install_clang_llvm(east: EastContext, exe_path: str):
+    """Installs clang+llvm to a proper location"""
+
+    clang_dir = os.path.join(east.consts["tooling_dir"], "clang+llvm")
+
+    # Remove old cppcheck dir if it exists
+    shutil.rmtree(clang_dir, ignore_errors=True)
+
+    east.print("[bold blue]Extracing clang+llvm binaries...")
+    shutil.unpack_archive(exe_path, east.consts["tooling_dir"], format="gztar")
+    east.print("[bold green]Done!")
+
+    # Remove version from the folder name
+    dir = glob.glob(os.path.join(east.consts["tooling_dir"], "clang+llvm-*"))[0]
+    shutil.move(dir, clang_dir)
```

Comment:
You use a folder name without a version for clang+llvm and a folder with a version for cppcheck. Is this intended? 

Are you meant to install multiple versions of cppcheck and only one version of clang?

Reply:
No, the issue is that the downloaded cppcheck tarball extracts into a folder with the name format `cppcheck-<version>`.

Line 170, performs rename.